Show Workload's JSON log lines in a better way

leandroberetta opened the pull request:
Resolves #7066.

jshaughn commented on frontend/cypress/integration/featureFiles/istio_config.feature:
```diff
@@ -13,7 +13,6 @@ Feature: Kiali Istio Config page
     And user selects the "bookinfo" namespace
 
   @bookinfo-app
-  @selected
```
That should not be there, it was probably mistakenly left over from another PR. `@selected` can be used to run only a selected subset of tests. It's very useful in development but should not be left in the code.

leandroberetta replied:
Well, I think it was merged by mistake, I guess we don't want to commit the @selected decorators. They are just for testing quickly the test that you're working on. On the CI, all tests are executed anyway, but when developing and testing this issue, I just want to run my new test, not the unrelated, and this Istio test were being executed too, so I think this might need to be removed.

leandroberetta commented:
How to test this PR:

- Install Kiali
- There is no configuration needed on the Kiali side
- Create a namespace for the dummy app which logs in json format
- Create the app
- Check the workloads logs

```
k create ns json-logger

kubectl run web-server-logger --image=busybox --restart=Never --namespace json-logger --command -- sh -c '
get_random_element() {
    local list_str="$1"
    set -- $list_str
    local num_elements=$#
    local random_index=$(( RANDOM % num_elements + 1 ))
    eval echo "\$$random_index"
}

while true; do
    METHODS_STR="GET POST PUT DELETE"
    PATHS_STR="/api/v1/users /api/v1/products /home /dashboard"
    STATUS_CODES_STR="200 201 404 500"
    USER_AGENTS_STR="\"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36\" \"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.5 Safari/605.1.15\" \"PostmanRuntime/7.39.0\""
    IP_ADDRESSES_STR="192.168.1.1 10.0.0.5 172.16.0.10"

    RANDOM_METHOD=$(get_random_element "$METHODS_STR")
    RANDOM_PATH=$(get_random_element "$PATHS_STR")
    RANDOM_STATUS=$(get_random_element "$STATUS_CODES_STR")
    RANDOM_USER_AGENT=$(get_random_element "$USER_AGENTS_STR")
    RANDOM_IP=$(get_random_element "$IP_ADDRESSES_STR")

    echo "{\"timestamp\":\"$(date -u +%Y-%m-%dT%H:%M:%SZ)\",\"level\":\"info\",\"service\":\"web-server\",\"event\":\"http_request\",\"request\":{\"method\":\"${RANDOM_METHOD}\",\"path\":\"${RANDOM_PATH}\",\"status\":${RANDOM_STATUS}},\"client\":{\"ip\":\"${RANDOM_IP}\",\"userAgent\":${RANDOM_USER_AGENT}},\"duration_ms\":$(( ( RANDOM % 500 ) + 50 ))}"
    sleep 1
    echo "Simplified log entry"
done
'
```

leandroberetta commented:
@josunect I addressed your comments:

![image](https://github.com/user-attachments/assets/ad24609b-ca77-444f-9c79-8807fb90fa86)

jshaughn reviewed (changes requested):
Nice addition. I think we can improve the dialog a little bit:
![image](https://github.com/user-attachments/assets/24cb44b0-265f-45af-b796-ce5b1adcf96a)

I'd suggest making the dialog larger, the log message will likely scroll. Perhaps we could use the same size as what we have in the parsed access log?
![image](https://github.com/user-attachments/assets/61e3f35a-9b0d-49f0-8a7c-ea0315713bee)

I think the parsed log dialogs should be the same size and this new dialog should be at least as large as the parsed access log dialog.

This would be a good time to improve consistency in both the existing and new dialog by adding the Close button. Your new dialog could even support Copy and Download of the formatted JSON?

Minor, can we make the title consistent with the parsed access log dialog?
![image](https://github.com/user-attachments/assets/57e0c2e5-3971-4cde-b9f3-cb6a5a65db13)

Maybe something like, "JSON Log Entry"

Also Minor, can we simplify the Tab titles to just "JSON" and "Table" , like what was shown in the issue description (although I like JSON on the left):
![image](https://github.com/user-attachments/assets/b31e7777-f0e9-43f4-815e-14cd0923a16c)

jshaughn commented:
Also, I tested to see what happens if proxy access logs are set for JSON encoding, and it works fine, so that's good!

Using `hack/istio/install-istio-via-istioctl.sh -c kubectl --set values.meshConfig.accessLogEncoding=JSON`

![image](https://github.com/user-attachments/assets/323bdbb4-50d0-432d-980d-ecd9359e539e)

leandroberetta commented:
> > @josunect I addressed your comments:
> > ![image](https://private-user-images.githubusercontent.com/1286393/454866131-ad24609b-ca77-444f-9c79-8807fb90fa86.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTAwNjk4NTcsIm5iZiI6MTc1MDA2OTU1NywicGF0aCI6Ii8xMjg2MzkzLzQ1NDg2NjEzMS1hZDI0NjA5Yi1jYTc3LTQ0NGYtOWM3OS04ODA3ZmI5MGZhODYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDYxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA2MTZUMTAyNTU3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODdjMzhjODBmNDNhYmE4MzA0NTI0ODEwYzIwMTJhMWQ0YjNiYTBkNTY5MzY2MTZjY2FlOWM4ZjRmNzgzZTJhNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.j4fW1Qkx0ml0BF9JKsjLavO-1Cmb9Nr3y6iJeo0rs88)
> 
> Thanks, @leandroberetta!
> 
> The approach I was following with UX was to establish also a height in the modal to avoid the resizing of it while changing from one tab to another.
> 
> I've installed the demo but I'm not able to see those log lines in your screenshot, not sure if they are intended to be also included in the PR?

Hi, I didn't added the following example to the demo, but you can install as I commented above:

```
kubectl create ns json-logger

kubectl run web-server-logger --image=busybox --restart=Never --namespace json-logger --command -- sh -c '
get_random_element() {
    local list_str="$1"
    set -- $list_str
    local num_elements=$#
    local random_index=$(( RANDOM % num_elements + 1 ))
    eval echo "\$$random_index"
}

while true; do
    METHODS_STR="GET POST PUT DELETE"
    PATHS_STR="/api/v1/users /api/v1/products /home /dashboard"
    STATUS_CODES_STR="200 201 404 500"
    USER_AGENTS_STR="\"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36\" \"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.5 Safari/605.1.15\" \"PostmanRuntime/7.39.0\""
    IP_ADDRESSES_STR="192.168.1.1 10.0.0.5 172.16.0.10"

    RANDOM_METHOD=$(get_random_element "$METHODS_STR")
    RANDOM_PATH=$(get_random_element "$PATHS_STR")
    RANDOM_STATUS=$(get_random_element "$STATUS_CODES_STR")
    RANDOM_USER_AGENT=$(get_random_element "$USER_AGENTS_STR")
    RANDOM_IP=$(get_random_element "$IP_ADDRESSES_STR")

    echo "{\"timestamp\":\"$(date -u +%Y-%m-%dT%H:%M:%SZ)\",\"level\":\"info\",\"service\":\"web-server\",\"event\":\"http_request\",\"request\":{\"method\":\"${RANDOM_METHOD}\",\"path\":\"${RANDOM_PATH}\",\"status\":${RANDOM_STATUS}},\"client\":{\"ip\":\"${RANDOM_IP}\",\"userAgent\":${RANDOM_USER_AGENT}},\"duration_ms\":$(( ( RANDOM % 500 ) + 50 ))}"
    sleep 1
    echo "Simplified log entry"
done
'
```

I will add the height to the modal.

josunect reviewed (commented):
Thanks, @leandroberetta ! I think it looks much better now. 
I've just added one comment.

jshaughn commented:
> There is a comment about a close button that I'm not sure if I understood it because the close button is there.

Sorry for the confusion. I was not referencing the top-right `X`, I was talking about adding the bottom buttons that are typically found using the PF guidelines. As an example, like on the Debug Information modal:

leandroberetta commented:
> > There is a comment about a close button that I'm not sure if I understood it because the close button is there.
> 
> Sorry for the confusion. I was not referencing the top-right `X`, I was talking about adding the bottom buttons that are typically found using the PF guidelines. As an example, like on the Debug Information modal: ![image](https://private-user-images.githubusercontent.com/2104052/456642042-27c2d75d-7349-4902-87fe-08a809952538.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTAzMzcxNTgsIm5iZiI6MTc1MDMzNjg1OCwicGF0aCI6Ii8yMTA0MDUyLzQ1NjY0MjA0Mi0yN2MyZDc1ZC03MzQ5LTQ5MDItODdmZS0wOGE4MDk5NTI1MzgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDYxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA2MTlUMTI0MDU4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MjZmOGY0YTYyMmFjN2ZmYzU3MmM1MDk4ZTk0Yzg4ODc5NzVmY2YyNzhhYzdjM2JkYTM1MDUxMTBlYmY4MGVmMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.ts4ARHwqMN4X34G0pcM8fLgDZEhjeqsSP3RXPcuodEs)

No problem, I see what you mean now, sure, I will add the close button alongside a copy button.
